### PR TITLE
fix(emails): Render email content right to left for rtl locales

### PR DESCRIPTION
**Pull request by Yogita98**
## Because

- Emails are rendered left to right by default

## This pull request

- Will render emails right to left for rtl locales and left to right for ltr locales

## Issue that this pull request solves

Closes: #10011 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)
Before:
Emails rendered ltr by default for all the locales
image

After:
Emails rendered rtl for ar (rtl) locale
image

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-auth-server/lib/senders/emails/renderer.ts
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/renderer.ts:11]: When the tests for scripts are being run, the `emails-scss` step doesn't get executed as part of the `write-emails-to-disk` test and hence the validation errors were being thrown. Executing the compilation step will ensure sass files are compiled down to css irrespective of from where its being called.

File: packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.ts
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.ts:22]: I had to create a separate file for rtl and ltr classes since we don't want these styles to be inlined. These styles are getting applied dynamically so either we do this way or render the styles from inside Js something like this:
```
if(isLocaleRenderedRtl){
      const head = document.getElementsByTagName('head')[0];
      const style = document.createElement("style");
      /* @ts-ignore */
      // style.type = 'text/css'
      style.innerHTML = `.rtl div {         
        direction: rtl !important;
       }`;
      head.appendChild(style);
    }
```

File: packages/fxa-auth-server/lib/senders/emails/templates/cadReminder/ar.ftl
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/templates/cadReminder/ar.ftl:1]: Even this file is added for testing purpose. I'll remove it when the Pr is mergeable

File: packages/fxa-auth-server/lib/senders/emails/renderer.ts
Comment (LZoog) [reply packages/fxa-auth-server/lib/senders/emails/renderer.ts:11]: Ah interesting, nice find. 🙂 I'm glad we know why CI was choking on this now.

I'm OK with leaving this here for now since this makes CI happy and this PR is about RTL styles if this isn't quick to change, but I think we can compile all of the SCSS to CSS at the auth-server build step. We don't need SCSS files except during development.

File: packages/fxa-auth-server/lib/senders/emails/rtl.scss
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/rtl.scss:1]: Since this also explicitly sets `ltr`, maybe we can call this `ltr-rtl.scss`, or `locale-dir.scss`?

**Review by LZoog — COMMENTED** [review]
This is looking really good! I really just had a few minor comments. 😄

Also, will you update your "closes" to close #10011? I scooted out #9210 into the last milestone since technically that won't be fixed until all FxA emails support this.

File: packages/fxa-auth-server/lib/senders/emails/fluent-localizer.ts
Comment (Yogita98) [reply packages/fxa-auth-server/lib/senders/emails/fluent-localizer.ts:30]: I think I googled the [list](https://meta.wikimedia.org/wiki/Template:List_of_language_names_ordered_by_code) of locales that are rtl or ltr, but we can definitely run this by someone in the FxA team.

File: packages/fxa-auth-server/lib/senders/emails/renderer.ts
Comment (Yogita98) [reply packages/fxa-auth-server/lib/senders/emails/renderer.ts:11]: Ahhh okay so it's okay if the scss are not compiled down to css during testing

**Review by jodyheavener — APPROVED** [review]
So I've previewed this and it works as described, but because my RTL knowledge is limited I'm not entirely sure how to push it to it's limits/test edge cases. Unless others find things I'm happy with how this looks 👍

File: packages/fxa-auth-server/lib/senders/emails/renderer.ts
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/renderer.ts:14]: I think this can just be `config.env === 'test'` since that evaluates to `true`/`false`.

File: packages/fxa-auth-server/lib/senders/emails/fluent-localizer.ts
Comment (LZoog) [reply packages/fxa-auth-server/lib/senders/emails/fluent-localizer.ts:30]: 👍 I've asked someone from the l10n team to look at this, we can tweak it later if they don't get a chance before we merge this PR. 🙂

**Comment by jodyheavener** [comment]
@Yogita98 I'm starting to think that failure in test-auth-server might be legit, let us know if you want another set of eyes on it!

**Comment by Yogita98** [comment]
Right, mjml is throwing validation errors again for not being able to read the css files. I tried to do what you suggested me,[ ignoring include files](https://github.com/Yogita98/fxa/blob/fix-emails-rtl/packages/fxa-auth-server/lib/senders/emails/renderer.ts#L14) when the tests are being run but aren't we using dev env for our [tests](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/scripts/test-local.sh#L8) as well?

File: packages/fxa-auth-server/lib/senders/emails/renderer.ts
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/renderer.ts:13]: Wasn't able to fix the CI test failures so turning this validation back to soft for now and we can circle back on the issue in a follow up PR

Comment (LZoog) [reply packages/fxa-auth-server/lib/senders/emails/renderer.ts:13]: That's fine by me, the `cadReminderFirst` ticket is that sort of catch-all for anything remaining/follow ups (plus after both briefly looking at it, we're not sure what the issue is 🤨), let's just be sure to note on that ticket that we need to address this. Jody gave this the r+ and it LGTM, so I'm going to merge it! 🎉

**Pull request merged**
